lib/Service/ProvisioningService.php

```diff
@@ -230,6 +235,19 @@ public function provisionUser(string $tokenUserId, int $providerId, object $idTo
 			$account->setProperty('phone', $event->getValue(), $scope, '1', '');
 		}
 
+		$event = new AttributeMappedEvent(ProviderService::SETTING_MAPPING_LANGUAGE, $idTokenPayload, $language);
+		$this->eventDispatcher->dispatchTyped($event);
+		$this->logger->debug('Phone mapping event dispatched');
+		if ($event->hasValue()) {
+			$language = $event->getValue();
+			$languagesCodes = $this->l10nFactory->findAvailableLanguages();
+			if (in_array($language, $languagesCodes, true) || $language === 'en') {
```

Comment:
Why is there a special case for en?

Reply:
Because I checked how the language is set by the settings app with `PUT /ocs/v2.php/cloud/users/USER_ID` : https://github.com/nextcloud/server/blob/master/apps/provisioning_api/lib/Controller/UsersController.php#L1080-L1084
to know which values are valid and did the same.